packages/website/blog/2024-03-01-parser-options-project-service.md

```diff
@@ -0,0 +1,298 @@
+- 🧠 **Predictability**: more closely aligns type information for lint rules to what editors produce
+
+This blog post will cover how `parserOptions.projectService` improves performance, simplifies configurations, and brings linting type information much closer to what editors such as VS Code run with.
+
+<!-- truncate -->
+
+## Introducing the Project Service
+
+Back in [Relative TSConfig Projects with `parserOptions.project = true` > Project Services](https://typescript-eslint.io/blog/parser-options-project-true#project-services), we'd mentioned we're working on a replacement for `parserOptions.project`:
+
+> The downside of having users specify `parserOptions.project` at all is that `@typescript-eslint/parser` needs manual logic to create TypeScript Programs and associate them with linted files.
+> Manual Program creation logic comes with a few issues:
+>
+> - Complex project setups can be difficult to get right.
+>   - For example, [typescript-eslint does not yet support Project References](https://github.com/typescript-eslint/typescript-eslint/issues/2094).
+> - The TypeScript compiler options used in the user's editor might differ from the compiler options in the TSConfigs they specified on disk.
+> - Files not included in created Programs can't be linted with type information, even though editors still typically surface type information when editing those files.
+>   - Most commonly, `.eslintrc.(c)js` files can be tricky to lint, resulting in the dreaded [_TSConfig does not include this file_ error](/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file).
+>
+> We're working on an option to instead call the same TypeScript "Project Service" APIs that editors such as VS Code use to create Programs for us instead.
+> Project Services will automatically detect the TSConfig for each file (like `project: true`), and will also allow type information to be computed for JavaScript files without the `allowJs` compiler option (unlike `project: true`).
+>
+> We hope this option will eventually become the standard way to enable typed linting.
+> However, because it's so new and untested, we're keeping it under the `EXPERIMENTAL_` prefix for all of the `6.??` versions.
+
+Following several months of discussion and testing, we believe the new Project Service API is ready to be used by real-world projects.
+We've found them to be generally faster at runtime and more straightforward to configure.
+
+We're therefore promoting the `parserOptions.EXPERIMENTAL_useProjectService` option to the stable **`parserOptions.projectService`** in typescript-eslint v7.
+
+## Onboarding to the Project Service
+
+You can change over to the new Project Service API by replacing `project` with `projectService` in your ESLint configuration:
+
+<!--tabs-->
+
+### Flat ESLint Config
+
+```diff title="eslint.config.js"
```

Comment:
nit you can use `js` for this and use the `// Remove this line` and `// Add this line` comments to mark the diff - this allows us to preserve syntax highlighting

eg https://typescript-eslint.io/linting/typed-linting/monorepos#one-tsconfigjson-per-package-and-an-optional-one-in-the-root
![screenshot of docs](https://github.com/typescript-eslint/typescript-eslint/assets/7462525/df2b2210-2d28-486c-b604-3647440f8d5a)